devito/interfaces.py

```diff
@@ -72,7 +73,20 @@ def __new__(cls, *args, **kwargs):
     def __init__(self):
         """Initialise from a cached instance by shallow copying __dict__."""
         original = _SymbolCache[self.__class__]
-        self.__dict__ = original.__dict__.copy()
+        self.__dict__ = original().__dict__.copy()
+
+    def clean(self):
+        """Cleans up a :class:`SymbolicData` object by removing
+        it from the symbolic cache.
+        """
+        if self._cached():
+            self._clean()
```

Comment:
Ok, now that we implement `__del__` I think we don't need to expose a `clean` function on the objects. Instead I would move this logic to `__del__`.